### ISSUE TITLE
"Source code property can't be modified" Upgrading from 1.2.0 to 1.2.1

### ISSUE THREAD

**Issue by phancox** [awslabs/aws-control-tower-customizations#36]
Updated the stack using the 1.2.1 template, no modification to parameters, and the update consistently fails with:

CustomControlTowerCodeCommit | UPDATE_FAILED | Source code property can't be modified.

Initial deployment used CodeCommit repo as the source.  aws-control-tower-customizations v1.1.0 I believe.

**Comment by groverlalit** [comment]
@phancox Thanks for bring this our attention. Could you please open a support case with more details to help us duplicate the issue? Thanks.

**Comment by phancox** [comment]
@groverlalit Did you mean an AWS support case under "Control Tower" or somewhere else?

**Comment by groverlalit** [comment]
@phancox Sorry for the confusion. For faster resolution let's continue on this channel. 
The CodeCommit as a source for Codepipeline was added in v1.2.0.

We tested upgrading from v1.2.0 to v1.2.1, Can you please confirm which version are you upgrading from? Are you trying to use an existing repo or trying to create a new one using the solution?

**Comment by phancox** [comment]
@groverlalit Going back through the stack event history:
- Stack was first created on 29-May-2020.  (Presumably version 1.1.0)
- CustomControlTowerCodeCommit was created on 23-Jul-2020 (Version 1.2.0 update)
- CustomControlTowerCodeCommit failed to update on 05-Oct-2020 (Version 1.2.1 update)

OK got it figured out.

Because I had no need to alter any of the parameters from the previous update I just accepted the defaults.  For "ExistingRepository" I needed to change from "No" to "Yes".

That sort of goes against my understanding of idempotency of stack updates.  For example if you pass a snapshot_id to an RDS stack, the stack doesn't go back to the snapshot if the parameter remains the same on the next stack update.  But if you change it to something else or remove it, the DB will recreate.

Anyway, all good now.

Thanks
